pkg/cmd/step/helm/step_helm_apply.go

```diff
@@ -313,6 +313,39 @@ func (o *StepHelmApplyOptions) Run() error {
 		return err
 	}
 
+	// Now let's unpack all the dependencies and apply the vault URLs
+	dependencies, err := filepath.Glob(filepath.Join(dir, "charts", "*.tgz"))
+	if err != nil {
+		return errors.Wrapf(err, "finding chart dependencies in %s", filepath.Join(dir, "charts"))
+	}
+	for _, src := range dependencies {
+		dest := filepath.Join(dir, "charts")
+		err := util.UnTargzAll(src, dest)
+		if err != nil {
+			return errors.Wrapf(err, "untarring %s to %s", src, dest)
+		}
+		err = os.Remove(src)
+		if err != nil {
+			return errors.Wrapf(err, "removing %s", src)
+		}
+		filepath.Walk(dest, func(path string, info os.FileInfo, err error) error {
+			if _, fileName := filepath.Split(path); fileName == helm.ValuesFileName {
+
+				newFiles, cleanup, err := helm.DecorateWithSecrets([]string{path}, secretURLClient)
+				defer cleanup()
+				if err != nil {
+					return errors.Wrapf(err, "decorating %s with secrets", path)
+				}
+				err = util.CopyFile(newFiles[0], path)
```

Comment:
This has the risk of leaving secrets into the repository. e.g. when it executes in the pipeline the value files will be resolved in place. There is a risk to commit the values in git in case a commit or a PR follows this operations. Not sure if the `charts` folder is ignored in git in the main chart.

Reply:
Good point, we need to always run this in a temp dir. I'll see if we can switch to making that non optional.

Reply:
I've made it so that we always use the temp dir